src/source/Ice/TurnConnection.c

```diff
@@ -96,8 +96,10 @@ STATUS freeTurnConnection(PTurnConnection* ppTurnConnection)
     }
 
     // shutdown control channel
-    CHK_LOG_ERR(connectionListenerRemoveConnection(pTurnConnection->pConnectionListener, pTurnConnection->pControlChannel));
-    CHK_LOG_ERR(freeSocketConnection(&pTurnConnection->pControlChannel));
+    if (pTurnConnection->pControlChannel) {
```

Comment:
Explicitly check against NULL - something like if (pTurnConnection->pControlChannel != NULL)

Reply:
I am also interested to know how this change helps in reducing CPU consumption.